io/post_fv3.F90

```diff
@@ -2138,164 +2335,641 @@ subroutine set_postvars_fv3(wrt_int_state,grid_id,mype,mpicomp)
-                  ptop(i,j) = arrayr42d(i,j)
-                  if( abs(arrayr42d(i,j)-fillValue) < small)  ptop(i,j) = spval
-                  if(ptop(i,j) <= 0.0) ptop(i,j) = spval
-                enddo
-              enddo
-            endif
-
-            ! inst cloud bottom pressure
-            if(trim(fieldname)=='prescnvclb') then
-              !$omp parallel do default(none) private(i,j) shared(jsta,jend,ista,iend,spval,pbot,arrayr42d,fillValue)
+            ! wetness
+            if(trim(fieldname)=='wetness') then
+              !$omp parallel do default(none) private(i,j) shared(jsta,jend,ista,iend,smstav,arrayr42d,fillvalue,spval)
               do j=jsta,jend
                 do i=ista, iend
-                  pbot(i,j) = arrayr42d(i,j)
-                  if( abs(arrayr42d(i,j)-fillValue) < small) pbot(i,j) = spval
-                  if(pbot(i,j) <= 0.0) pbot(i,j) = spval
+                  smstav(i,j) = arrayr42d(i,j)
+                  if(abs(arrayr42d(i,j)-fillvalue)<small) smstav(i,j) = spval
                 enddo
               enddo
             endif
 
-            ! time averaged low cloud top pressure
-            if(trim(fieldname)=='pres_avelct') then
-              !$omp parallel do default(none) private(i,j) shared(jsta,jend,ista,iend,spval,ptopl,arrayr42d,fillValue)
+            if(modelname=='FV3R')then
+            !acsnow
+            if(trim(fieldname)=='accswe_land') then
+              !$omp parallel do default(none) private(i,j) shared(jsta,jend,ista,iend,buf,arrayr42d,fillvalue,spval)
               do j=jsta,jend
                 do i=ista, iend
-                  ptopl(i,j) = arrayr42d(i,j)
-                  if( abs(arrayr42d(i,j)-fillValue) < small) ptopl(i,j) = spval
+                  buf(i,j) = arrayr42d(i,j)
+                  if(abs(arrayr42d(i,j)-fillvalue)<small) buf(i,j) = spval
                 enddo
               enddo
             endif
-
-            ! time averaged low cloud bottom pressure
-            if(trim(fieldname)=='pres_avelcb') then
-              !$omp parallel do default(none) private(i,j) shared(jsta,jend,ista,iend,spval,pbotl,arrayr42d,fillValue)
+            if(trim(fieldname)=='accswe_ice') then
+              !$omp parallel do default(none) private(i,j) shared(jsta,jend,ista,iend,buf2,arrayr42d,fillvalue,spval)
               do j=jsta,jend
                 do i=ista, iend
-                  pbotl(i,j) = arrayr42d(i,j)
-                  if( abs(arrayr42d(i,j)-fillValue) < small) pbotl(i,j) = spval
+                  buf2(i,j) = arrayr42d(i,j)
+                  if(abs(arrayr42d(i,j)-fillvalue)<small) buf2(i,j) = spval
                 enddo
               enddo
             endif
+            ! !$omp parallel do default(none) private(i,j) shared(jsta,jend,ista,iend,spval,buf,buf2,acsnow)
+             do j=jsta,jend
+               do i=ista, iend
+                 if(buf(i,j)<spval .and. buf2(i,j)<spval) then
+                   acsnow(i,j) = buf(i,j) + buf2(i,j)
```

Comment:
What are the values in buf and buf2 arrays? I guess they should be `accswe_land` and `accswe_ice`. If I understand correctly this i,j loop will be executed for every iteration of `do n=1, ncount_field` loop since it's not inside an if test that checks the value of `fieldname`. Is that correct?

Reply:
@DusanJovic-NOAA That's right. Do you have any suggestions on this?
Adding if(trim(fieldname)=='accswe_ice' .and. trim(fieldname)=='accswe_ice') before i,j loop for calculating acsnow?

Reply:
That if test will not work because fieldname can not be both `accswe_ice` and `accswe_ice` at the same time.

Also this logic depends on the order of the fields in the bundle, which I do not think is very robust. What if `accswe_ice` and `accswe_ice` are not next to each other and for whatever reason other fields that also assign the values to buf or buf2, like `snacc_ice` are in between these two.

Reply:
Ideally this computation should be done somewhere in the model, not here. This routine should only copy data from 'model arrays' to 'post arrays', nothing else.

Reply:
> That if test will not work because fieldname can not be both `accswe_ice` and `accswe_ice` at the same time.
> 
> Also this logic depends on the order of the fields in the bundle, which I do not think is very robust. What if `accswe_ice` and `accswe_ice` are not next to each other and for whatever reason other fields that also assign the values to buf or buf2, like `snacc_ice` are in between these two.

@DusanJovic-NOAA You are right. How about moving calculation of acsnow and andepac outside of ncount_field loop (e.g. after line 4656) and also allocate another two bufs for snacc_land and snacc_ice?

Reply:
> Ideally this computation should be done somewhere in the model, not here. This routine should only copy data from 'model arrays' to 'post arrays', nothing else.

@DusanJovic-NOAA I will pass your comment to RRFS developers from GSL for future enhancement.

Reply:
> > That if test will not work because fieldname can not be both `accswe_ice` and `accswe_ice` at the same time.
> > Also this logic depends on the order of the fields in the bundle, which I do not think is very robust. What if `accswe_ice` and `accswe_ice` are not next to each other and for whatever reason other fields that also assign the values to buf or buf2, like `snacc_ice` are in between these two.
> 
> @DusanJovic-NOAA You are right. How about moving calculation of acsnow and andepac outside of ncount_field loop (e.g. after line 4656) and also allocate another two bufs for snacc_land and snacc_ice?

That should work. And instead of buf and buf2 please use arrays named accswe_land, accswe_ice, snacc_land, snacc_ice.

Reply:
You may want to use two logic 2D variables to mark if the two fields are set up from the model. Also I think final fields can't be set up as: ice+land because the two fields can't overlap with each other, you may end up with the full field values >spval. See line 2356 and 2536.

Reply:
To prevent these kid of bugs in the future I think almost the entire `ncount_field` loop should be rewritten using 
```

if(trim(fieldname)==......) then

else if(trim(fieldname)==......) then

end if
```

or select case statement instead of separate if tests

Also there is a code at the end of  `ncount_field` loop that is executed for every field, for example loop that computes PM concentrations. I do not think that should be done in a loop that is executed hundreds of times.